src/platform/forms-system/src/js/components/FormSignature.jsx

```diff
@@ -0,0 +1,183 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { setData } from 'platform/forms-system/src/js/actions';
+import set from 'platform/utilities/data/set';
+import TextInput from '@department-of-veterans-affairs/component-library/TextInput';
+import Checkbox from '@department-of-veterans-affairs/component-library/Checkbox';
+
+/**
+ * Description of how the component behaves here.
+ *
+ * Example usage in formConfig:
+ * preSubmitInfo: {
+ *   CustomComponent: (signatureProps) => (
+ *     <section className="box vads-u-background-color--gray-lightest vads-u-padding-bottom--6 vads-u-padding-x--3 vads-u-padding-top--1px vads-u-margin-bottom--7">
+ *       <h3>Statement of truth</h3>
+ *       <p>I solemnly swear I am up to no good.</p>
+ *       <FormSignature
+ *         {...signatureProps}
+ *         signatureLabel="Secret code name"
+ *       />
+ *     <section/>
+ *   );
+ * }
+ */
+export const FormSignature = ({
+  signatureLabel,
+  checkboxLabel,
+  formData,
+  setFormData,
+  signaturePath,
+  required,
+  showError,
+  validations,
+  onSectionComplete,
+}) =>
+  // {
+  //   formData,
+  //   onSectionComplete,
+  //   setFormData,
+  //   showError,
+  //   submission,
+  // },
+  {
+    // Input states
+    const [signature, setSignature] = useState({ value: '', dirty: false });
+    const [checked, setChecked] = useState(false);
+
+    // Validation states
+    const [signatureError, setSignatureError] = useState(null);
+    const [checkboxError, setCheckboxError] = useState(null);
+
+    // Section complete state (so callback isn't called too many times)
+    const [sectionComplete, setSectionComplete] = useState(false);
+
+    // Signature input validation
+    useEffect(
+      () => {
+        // Required validation always comes first
+        if (required)
+          validations.unshift(
+            signatureValue =>
+              !signatureValue ? 'Please sign your name' : undefined,
```

Comment:
@peggygannon What should this default error message be?

**Context:** This will show up only if `required` is set to true. It'll only check that the text field has _something_ in it. It won't check that the name actually matches anything, so I thought we may not want to say `Your signature must match your first and last name as previously entered` for this particular message. ([The original design](https://app.zeplin.io/project/5e29e7ff922d4d961b5bff4d/screen/6092a7ba7e59383820819a3e) had this language.) You can see the screenshot above for an example of what it looks like.